console/src/app/pages/projects/apps/integrate/integrate.component.ts

```diff
@@ -0,0 +1,204 @@
+    private router: Router,
+    private toast: ToastService,
+    private dialog: MatDialog,
+    private mgmtService: ManagementService,
+    private _location: Location,
+    private breadcrumbService: BreadcrumbService,
+    public navigation: NavigationService,
+  ) {
+    effect(() => {
+      const fwId = this.framework()?.id;
+      const fw = this.framework();
+      if (fw && fwId) {
+        const request = OIDC_CONFIGURATIONS[fwId];
+        request.setProjectId(this.projectId);
+        request.setName(fw.title);
+        request.setDevMode(true);
+        this.requestRedirectValuesSubject$.next();
+
+        this.oidcAppRequest.next(request);
+        return request;
+      } else {
+        const request = new AddOIDCAppRequest();
+        this.oidcAppRequest.next(request);
+        return request;
+      }
+    });
+  }
+
+  public projectName$ = combineLatest([this.mgmtService.ownedProjects, this.mgmtService.grantedProjects]).pipe(
+    map(([projects, grantedProjects]) => {
+      const project = projects.find((project) => project.id === this.activatedRoute.snapshot.paramMap.get('projectid'));
+
+      const grantedproject = grantedProjects.find(
+        (grantedproject) => grantedproject.projectId === this.activatedRoute.snapshot.paramMap.get('projectid'),
+      );
+
+      return project?.name ?? grantedproject?.projectName ?? '';
+    }),
+  );
+
+  public setFramework(framework: Framework | undefined) {
+    this.framework.set(framework);
+  }
+
+  public ngOnInit(): void {
+    const projectId = this.activatedRoute.snapshot.paramMap.get('projectid');
+    if (projectId) {
+      const breadcrumbs = [
+        new Breadcrumb({
+          type: BreadcrumbType.ORG,
+          routerLink: ['/org'],
+        }),
+        new Breadcrumb({
+          type: BreadcrumbType.PROJECT,
+          name: '',
+          param: { key: 'projectid', value: projectId },
+          routerLink: ['/projects', projectId],
+          isZitadel: false,
+        }),
+      ];
+      this.projectId = projectId;
+      this.breadcrumbService.setBreadcrumb(breadcrumbs);
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  public close(): void {
+    if (this.navigation.isBackPossible) {
+      this._location.back();
+    } else {
+      this.router.navigate(['/projects', this.projectId]);
+    }
+  }
+
+  public createApp(): void {
+    this.loading = true;
+    this.mgmtService
+      .addOIDCApp(this.oidcAppRequest.getValue())
+      .then((resp) => {
+        this.loading = false;
+        this.toast.showInfo('APP.TOAST.CREATED', true);
+        if (resp.clientSecret) {
+          this.showSavedDialog(resp);
+        } else {
+          this.router.navigate(['projects', this.projectId, 'apps', resp.appId], { queryParams: { new: true } });
+        }
+      })
+      .catch((error) => {
+        this.loading = false;
+        this.toast.showError(error);
```

Comment:
I think the already exists error is a bit misleading. It suggests you can only have a single app of a framework in the same project.

As a quickfix, we could catch this error here and retry to create the app using a random suffix in the name. Or even better, show a dialog that asks the user to specify a different name and then retry to create the app. What do you think about this?

Reply:
IMO we also don't want to create duplicate apps, so this would prevent that.

Reply:
Do you have another suggestion on how we can ensure that a user doesn't think that he can have only one vue app in his project? I imagine if this happens, it has potential for frustration.
Maybe I'm wrong and this use case is too exotic, but :man_shrugging:

Reply:
To handle this error gracefully, we should catch the error on the /integrate page and also allow to edit the name there. I would not create the app with a random suffix though 🤔